lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/api/ResourceRequestEventImpl.java

```diff
@@ -47,6 +50,12 @@ private ResourceRequestEventImpl(final Builder builder) {
     this.virtualCores = Optional.ofNullable(builder.virtualCores);
     this.relaxLocality = Optional.ofNullable(builder.relaxLocality);
     this.runtimeName = builder.runtimeName == null ? "" : builder.runtimeName;
+    this.nodeLabels = builder.nodeLabels == null ? new HashMap<String, String>() : builder.nodeLabels;
+  }
+
+  @Override
+  public Map<String, String> getNodeLabels() {
+    return nodeLabels;
```

Comment:
Same as above: This hands a reference to a mutable collection to the caller. Please explain if and why this is safe.